contracts/implementation/PoolCommitter.sol

```diff
@@ -59,301 +47,251 @@ contract PoolCommitter is IPoolCommitter, Ownable {
         uint256 updateInterval = pool.updateInterval();
         uint256 lastPriceTimestamp = pool.lastPriceTimestamp();
         uint256 frontRunningInterval = pool.frontRunningInterval();
 
+        Commitment storage totalCommit;
+        Commitment storage userCommit;
+
         if (
             PoolSwapLibrary.isBeforeFrontRunningInterval(
-                lastQueueLengthReset,
-                lastPriceTimestamp,
-                updateInterval,
-                frontRunningInterval
-            ) &&
-            !PoolSwapLibrary.isBeforeFrontRunningInterval(
                 block.timestamp,
                 lastPriceTimestamp,
                 updateInterval,
                 frontRunningInterval
             )
         ) {
-            /**
-             * The lastQueueLengthReset occured before the frontRunningInterval,
-             * and we are within the frontRunningInterval,
-             * so this is the first commit since frontRunningInterval has passed.
-             * Note: If and only if there are no `commit` calls within the frontRunningInterval, then
-             * `executeAllCommitments` will reset `currentCommitQueueLength` and update
-             * `lastQueueLengthReset`.
-             */
-            delete currentCommitQueueLength;
-            lastQueueLengthReset = block.timestamp;
+            totalCommit = totalMostRecentCommit;
+            userCommit = userMostRecentCommit[msg.sender];
+            userCommit.updateIntervalId = updateIntervalId;
+        } else {
+            totalCommit = totalNextIntervalCommit;
+            userCommit = userNextIntervalCommit[msg.sender];
+            userCommit.updateIntervalId = updateIntervalId + 1;
         }
-        currentCommitQueueLength += 1;
-        uint128 currentCommitIDCounter = commitIDCounter;
-        commitIDCounter = currentCommitIDCounter + 1;
-
-        // create commitment
-        commits[currentCommitIDCounter] = Commit({
-            commitType: commitType,
-            amount: amount,
-            owner: msg.sender,
-            created: uint40(block.timestamp)
-        });
-        uint256 _commitType = uint256(commitType);
-        shadowPools[_commitType] = shadowPools[_commitType] + amount;
-
-        if (earliestCommitUnexecuted == NO_COMMITS_REMAINING) {
-            earliestCommitUnexecuted = currentCommitIDCounter;
-        }
-        latestCommitUnexecuted = currentCommitIDCounter;
-
-        emit CreateCommit(currentCommitIDCounter, amount, commitType);
-        uint256 shortBalance = pool.shortBalance();
-        uint256 longBalance = pool.longBalance();
 
-        // pull in tokens
         if (commitType == CommitType.LongMint || commitType == CommitType.ShortMint) {
             // minting: pull in the quote token from the committer
-            require(amount >= minimumCommitSize, "Amount less than minimum");
             pool.quoteTokenTransferFrom(msg.sender, leveragedPool, amount);
```

Comment:
why not add this in the big if statement below to the CommitType.LongMint and CommitType.ShortMint sections and remove this if statement.